src/fileformat/file_format/pe/pe_format.cpp

```diff
@@ -622,6 +626,544 @@ void PeFormat::loadRichHeader()
+	std::vector<std::uint8_t> bytes;
+	struct VBExternTableEntry entry;
+	struct VBExternTableEntryData entryData;
+	unsigned long long vbExternEntryDataOffset = 0;
+	std::size_t offset = 0;
+
+	for (std::size_t i = 0; i < nEntries; i++)
+	{
+		std::string moduleName;
+		std::string apiName;
+
+		if (!getBytes(bytes, structureOffset + i * entry.structureSize(), entry.structureSize())
+			|| bytes.size() != entry.structureSize())
+		{
+			break;
+		}
+
+		offset = 0;
+		DynamicBuffer entryContent(bytes, retdec::utils::Endianness::LITTLE);
+		entry.type = entryContent.read<std::uint32_t>(offset); offset += sizeof(entry.type);
+		entry.importDataAddr = entryContent.read<std::uint32_t>(offset); offset += sizeof(entry.importDataAddr);
+
+		if (entry.type != static_cast<std::uint32_t>(VBExternTableEntryType::external))
+		{
+			continue;
+		}
+
+		if (!getOffsetFromAddress(vbExternEntryDataOffset, entry.importDataAddr))
+		{
+			continue;
+		}
+
+		if (!getBytes(bytes, vbExternEntryDataOffset, entryData.structureSize()) 
+			|| bytes.size() != entryData.structureSize())
+		{
+			continue;
+		}
+
+		offset = 0;
+		DynamicBuffer entryDataContent(bytes, retdec::utils::Endianness::LITTLE);
+		entryData.moduleNameAddr = entryDataContent.read<std::uint32_t>(offset); offset += sizeof(entryData.moduleNameAddr);
+		entryData.apiNameAddr = entryDataContent.read<std::uint32_t>(offset); offset += sizeof(entryData.apiNameAddr);
+
+		unsigned long long moduleNameOffset;
+		if (getOffsetFromAddress(moduleNameOffset, entryData.moduleNameAddr))
+		{
+			moduleName = retdec::utils::readNullTerminatedAscii(allBytes.data(), allBytes.size(),
+														moduleNameOffset, VB_MAX_STRING_LEN, true);
+		}
+
+		unsigned long long apiNameOffset;
+		if (getOffsetFromAddress(apiNameOffset, entryData.apiNameAddr))
+		{
+			apiName = retdec::utils::readNullTerminatedAscii(allBytes.data(), allBytes.size(),
+														apiNameOffset, VB_MAX_STRING_LEN, true);
+		}
+
+		if (!moduleName.empty() || !apiName.empty())
+		{
+			auto ext = std::make_unique<VisualBasicExtern>();
+			ext->setModuleName(moduleName);
+			ext->setApiName(apiName);
+			visualBasicInfo.addExtern(std::move(ext));
+		}
+	}
+
+	visualBasicInfo.computeExternTableHashes();
+
+	return true;
+}
+
+/**
+ * Parse visual basic object table
+ * @param structureOffset Offset in file where the structure starts
+ * @return @c true if object table was successfuly parsed, @c false otherwise
+ */
+bool PeFormat::parseVisualBasicObjectTable(std::size_t structureOffset)
+{
+	const auto &allBytes = getBytes();
+	std::vector<std::uint8_t> bytes;
+	std::size_t offset = 0;
+	unsigned long long projectNameOffset = 0;
+	unsigned long long objectDescriptorsOffset = 0;
+	struct VBObjectTable vbot;
+	std::string projName;
+
+	if (!getBytes(bytes, structureOffset, vbot.structureSize()) || bytes.size() != vbot.structureSize())
+	{
+		return false;
+	}
+
+	DynamicBuffer structContent(bytes, retdec::utils::Endianness::LITTLE);
```

Comment:
Shouldn't endianness be the endianness of the file itself? This seems like it forces little endian on the data.

Reply:
PE is always LE ( https://reverseengineering.stackexchange.com/questions/17922/determining-endianness-of-pe-files-windows-on-arm )